02-peer-protocol.md

```diff
@@ -560,6 +1077,337 @@ ones it has sent so it can use them should they be requested by
+It changes the previous protocol to allow the `accept_channel2` peer
+(the *accepter*/*non-initiator*) to contribute inputs to the funding
+transaction, via the interactive transaction construction protocol.
+
+        +-------+                              +-------+
+        |       |--(1)--- open_channel2  ----->|       |
+        |       |<-(2)--- accept_channel2 -----|       |
+        |       |                              |       |
+    --->|       |      <tx collaboration>      |       |
+    |   |       |                              |       |
+    |   |       |--(3)--  commitment_signed -->|       |
+    |   |       |<-(4)--  commitment_signed ---|       |
+    |   |   A   |                              |   B   |
+    |   |       |<-(5)--  tx_signatures -------|       |
+    |   |       |--(6)--  tx_signatures ------>|       |
+    |   |       |                              |       |
+    |   |       |--(a)--- tx_init_rbf -------->|       |
+    ----|       |<-(b)--- tx_ack_rbf ----------|       |
+        |       |                              |       |
+        |       |    <tx rbf collaboration>    |       |
+        |       |                              |       |
+        |       |--(c)--  commitment_signed -->|       |
+        |       |<-(d)--  commitment_signed ---|       |
+        |       |                              |       |
+        |       |<-(e)--  tx_signatures -------|       |
+        |       |--(f)--  tx_signatures ------>|       |
+        |       |                              |       |
+        |       |--(7)--- channel_ready  ----->|       |
+        |       |<-(8)--- channel_ready  ------|       |
+        +-------+                              +-------+
+
+        - where node A is *opener*/*initiator* and node B is
+          *accepter*/*non-initiator*
+
+### The `open_channel2` Message
+
+This message initiates the v2 channel establishment workflow.
+
+1. type: 64 (`open_channel2`)
+2. data:
+   * [`chain_hash`:`chain_hash`]
+   * [`channel_id`:`temporary_channel_id`]
+   * [`u32`:`funding_feerate_perkw`]
+   * [`u32`:`commitment_feerate_perkw`]
+   * [`u64`:`funding_satoshis`]
+   * [`u64`:`dust_limit_satoshis`]
+   * [`u64`:`max_htlc_value_in_flight_msat`]
+   * [`u64`:`htlc_minimum_msat`]
+   * [`u16`:`to_self_delay`]
+   * [`u16`:`max_accepted_htlcs`]
+   * [`u32`:`locktime`]
+   * [`point`:`funding_pubkey`]
+   * [`point`:`revocation_basepoint`]
+   * [`point`:`payment_basepoint`]
+   * [`point`:`delayed_payment_basepoint`]
+   * [`point`:`htlc_basepoint`]
+   * [`point`:`first_per_commitment_point`]
+   * [`point`:`second_per_commitment_point`]
+   * [`byte`:`channel_flags`]
+   * [`opening_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `opening_tlvs`
+2. types:
+   1. type: 0 (`upfront_shutdown_script`)
+   2. data:
+       * [`...*byte`:`shutdown_scriptpubkey`]
+   1. type: 1 (`channel_type`)
+   2. data:
+        * [`...*byte`:`type`]
+   1. type: 2 (`require_confirmed_inputs`)
```

Comment:
If `require_confirmed_inputs` is present, would it make sense to include block hash or height for added UTXOs? It would make it practical for light clients to uphold.

Reply:
Can you detail? Who would that help exactly, and how? IMO if you're a light client you simply won't require confirmed inputs from your LSP, who wouldn't accept that requirement anyway. This is rather the other way around, the node with a large volume of channel opens is the one that may require its peers to use confirmed inputs.

Reply:
Just thinking if it's reasonable to assume that maybe two light clients want to open a channel with one another. One of the peers (or both) wants to ensure the other uses confirmed inputs (to avoid paying fees for low feerate ancestors) but might not have an efficient way to query the UTXO set for the counterparty's UTXO, but with a block hash/height it could query and look for inclusion.

Sorry, I probably didn't understand the original reasoning for `require_confirmed_inputs`.

Reply:
> but might not have an efficient way to query the UTXO set for the counterparty's UTXO, but with a block hash/height it could query and look for inclusion.

If I'm understanding this correctly, this block hash / block height should be added to `tx_add_input`, right? You'd like every input to optionally specify the height/hash at which it was confirmed to help the receiver check that input's confirmation status?

That could be reasonable, but I don't think it makes sense in the base dual funding specification, because I'm not sure nodes would want to be forced to provide this information. In our case for example obtaining that information for our own inputs requires an additional call to `bitcoind`, which is annoying. So maybe that's something that could be added later (if we really need it for some use cases) with a dedicated feature bit (which some nodes wouldn't enable if it's costly for them to get this information).

Reply:
Yeah exactly that.

I don't think it's essential to have to get this merged, but I do feel it would be useful.

Happy to defer to a future feature. 👍